nemo/collections/asr/models/sortformer_diar_models.py

```diff
@@ -0,0 +1,578 @@
+            drop_last=config.get('drop_last', False),
+            shuffle=False,
+            num_workers=config.get('num_workers', 1),
+            pin_memory=config.get('pin_memory', False),
+        )
+        logging.info(f"AAC: Dataloader Instance loading is done ETA Step B done: {time.time() - time_flag}")
+        return dataloader_instance
+
+    def setup_training_data(self, train_data_config: Optional[Union[DictConfig, Dict]]):
+        self._train_dl = self.__setup_dataloader_from_config(
+            config=train_data_config,
+        )
+
+    def setup_validation_data(self, val_data_layer_config: Optional[Union[DictConfig, Dict]]):
+        self._validation_dl = self.__setup_dataloader_from_config(
+            config=val_data_layer_config,
+        )
+
+    def setup_test_data(self, test_data_config: Optional[Union[DictConfig, Dict]]):
+        self._test_dl = self.__setup_dataloader_from_config(
+            config=test_data_config,
+        )
+
+    def test_dataloader(self):
+        if self._test_dl is not None:
+            return self._test_dl
+        return None
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        if hasattr(self.preprocessor, '_sample_rate'):
+            audio_eltype = AudioSignal(freq=self.preprocessor._sample_rate)
+        else:
+            audio_eltype = AudioSignal()
+        return {
+            "audio_signal": NeuralType(('B', 'T'), audio_eltype),
+            "audio_signal_length": NeuralType(('B',), LengthsType()),
+        }
+
+    @property
+    def output_types(self) -> Dict[str, NeuralType]:
+        return OrderedDict(
+            {
+                "preds": NeuralType(('B', 'T', 'C'), ProbsType()),
+            }
+        )
+
+    def frontend_encoder(self, processed_signal, processed_signal_length):
+        """
+        Generate encoder outputs from frontend encoder.
+
+        Args:
+            processed_signal (torch.Tensor): tensor containing audio-feature (mel spectrogram, mfcc, etc.)
+            processed_signal_length (torch.Tensor): tensor containing lengths of audio signal in integers
+
+        Returns:
+            emb_seq (torch.Tensor): tensor containing encoder outputs
+            emb_seq_length (torch.Tensor): tensor containing lengths of encoder outputs
+        """
+        # Spec augment is not applied during evaluation/testing
+        if self.spec_augmentation is not None and self.training:
+            processed_signal = self.spec_augmentation(input_spec=processed_signal, length=processed_signal_length)
+        self.encoder = self.encoder.to(self.device)
+        emb_seq, emb_seq_length = self.encoder(audio_signal=processed_signal, length=processed_signal_length)
+        emb_seq = emb_seq.transpose(1, 2)
+        if self._cfg.encoder.d_model != self._cfg.tf_d_model:
+            self.sortformer_modules.encoder_proj = self.sortformer_modules.encoder_proj.to(self.device)
```

Comment:
don;t move to device at each step. setup before.

Reply:
changed